File: decode.py
num =input()
num = num[::-1]
Final_string = '' 
number = '' 
for i in range(len(num)): 
    number = number + num[i] 
    if (int(number) >= 65 and int(number) <= 90) or (int(number) >= 97 and 
    int(number) <= 122) or (int(number) == 32): 
      character =chr(int(number)) 
      Final_string+=character 
      number = '' 
      print(Final_string)

File: decode.js
const input = "23511011501782351112179911801562340161171141148";
let reversedString = "";

// Reverse the string
for (let i = input.length - 1; i >= 0; i--) {
  reversedString += input[i];
}

let decodedString = "";
let number = "";

for (let i = 0; i < reversedString.length; i++) {
  number += reversedString[i];
  if (
    (parseInt(number) >= 65 && parseInt(number) <= 90) ||
    (parseInt(number) >= 97 && parseInt(number) <= 122) ||
    parseInt(number) === 32
  ) {
    const character = String.fromCharCode(parseInt(number));
    decodedString += character;
    number = "";

  }
}

console.log(decodedString);